lib/k8y/rest/auth/token_store.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module K8y
+  module REST
+    module Auth
+      class TokenStore
+        class << self
+          def [](hostname)
+            lock.synchronize { store[hostname] }
+          end
```

Comment:
I don't like this (singleton class). I'd prefer to set this up to be an instantiable object to make working in different environments easier